src/main/java/org/verdictdb/jdbc41/VerdictStatement.java

```diff
@@ -35,11 +43,82 @@
 
   VerdictSingleResult result;
 
+  private VerdictDBLogger log = VerdictDBLogger.getLogger(this.getClass());
+
   public VerdictStatement(Connection conn, VerdictContext context) {
     this.conn = conn;
     this.executionContext = context.createNewExecutionContext();
   }
 
+  public ResultSet sql(String query) throws VerdictDBException {
+    VerdictStreamResultSet resultSet = new VerdictStreamResultSet();
+    if (getStreamQueryIfExists(query).getKey()) {
+      query = query.replaceFirst("(?i)stream", "");
+      VerdictResultStream resultStream = executionContext.streamsql(query);
+      new Thread(new ExecuteStream(resultStream, resultSet)).start();
+      return resultSet;
+    }
+    return null;
+  }
+
+  /**
+   * Created by Shucheng Zhong on 9/13/18
+   * <p>
+   * It will try to get the VerdictSingleResult from VerdictResultStream
+   * and append the VerdictSingleResult to VerdictStreamResultSet
+   * until all the VerdictSingleResults have return.
+   * Since it running on separate thread, it won't block user querying from
+   * the ResultSet
+   */
+  class ExecuteStream implements Runnable {
+
+    VerdictResultStream resultStream;
+
+    VerdictStreamResultSet resultSet;
+
+    ExecuteStream(VerdictResultStream resultStream, VerdictStreamResultSet resultSet) {
+      this.resultStream = resultStream;
+      this.resultSet = resultSet;
+    }
+
+    /**
+     * When the last SingleResult is returned, it will synchronized the isCompleted flag of VerdictStreamResultSet
+     * to block possible call of next(). It will set status of VerdictStreamResultSet to be complete and
+     * append the last SingleResult to VerdictStreamResultSet.
+     */
+    public void run() {
+      while (!resultStream.isCompleted()) {
+        while (resultStream.hasNext()) {
+          VerdictSingleResult singleResult = resultStream.next();
+          if (!resultStream.hasNext()) {
+            synchronized ((Object) resultSet.isCompleted) {
+              resultSet.appendSingleResult(singleResult);
+              resultSet.setCompleted();
+            }
+            log.debug("Execution Completed\n");
+          } else {
+            resultSet.appendSingleResult(singleResult);
+          }
+        }
+      }
+
+    }
+  }
+
+  private Pair<Boolean, String> getStreamQueryIfExists(String query) {
```

Comment:
Can you add a comment to describe what happens when the query does not include the "stream" keyword?

Reply:
It will return false. However, I think I should integrate this part to ExecutionContext.identifyQueryType().